collect_app/src/main/java/org/odk/collect/android/preferences/utielities/KnownUrlListUtils.java

```diff
@@ -0,0 +1,44 @@
+package org.odk.collect.android.preferences.utielities;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.preferences.GeneralSharedPreferences;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class KnownUrlListUtils {
+    static final String KNOWN_URL_LIST = "knownUrlList";
+
+    private KnownUrlListUtils() {
+    }
+
+    public static void addUrlToList(String url) {
+        List<String> urlList = getUrlList();
+
+        if (urlList.contains(url)) {
+            urlList.remove(url);
+        } else if (urlList.size() == 5) {
+            if (urlList.get(4).equals(Collect.getInstance().getString(R.string.default_server_url))) {
```

Comment:
This ensures that the default server URL is always preserved in the list, right? And that is new behavior? That doesn't seem right to me -- if the default URL hasn't been used, I think it makes sense to replace it.

Reply:
Yes, I don't know why but I was sure it worked in that way before, fixed.